lib/dead_end/api.rb

```diff
@@ -0,0 +1,191 @@
+require_relative "version"
+
+require "tmpdir"
+require "stringio"
+require "pathname"
+require "ripper"
+require "timeout"
+
+module DeadEnd
+  # Used to indicate a default value that cannot
+  # be confused with another input.
+  DEFAULT_VALUE = Object.new.freeze
+
+  class Error < StandardError; end
+  TIMEOUT_DEFAULT = ENV.fetch("DEAD_END_TIMEOUT", 1).to_i
+
+  # DeadEnd.handle_error [Public]
+  #
+  # Takes an exception from a syntax error, uses that
```

Comment:
Is the interface specifically that callers *must* pass a `SyntaxError` exception? I see that you don't mention that in the `rescue` line below, though you do mention it in the `lib/dead_end/core_ext.rb` file.

If it's intended for that requirement to be explicit, do you want to check that in `handle_error` like

```ruby
  raise TypeError.new("must pass a SyntaxError") unless e.is_a?(SyntaxError)
```

(I find that it's usually more forgiving to start with a narrow API and widen it, but it's up to you.)

Reply:
I'm a little torn. I agree that being stricter is better. Rather than raising a new error though I decided to warn and re-raise the original error which is a pattern I have in other places.

Take a look and let me know what you think. Also as a heads up, if you all haven't run into it yet `require_relative` monkeypatching is broken with Ruby 3.1 and needs to be fixed https://github.com/Shopify/bootsnap/commit/0d64e7e5e493884dc6a23a3ac4632d069172c4a1

Reply:
works for me.